modules/basic-skills/src/views/full/callApi.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import { Row, Col, Label, Input } from 'reactstrap'
+import Select from 'react-select'
+import style from './style.scss'
+import { BotpressTooltip } from 'botpress/tooltip'
+import { LinkDocumentationProvider } from 'botpress/documentation'
+
+const methodOptions = [
+  { label: 'get', value: 'get' },
+  { label: 'post', value: 'post' },
+  { label: 'put', value: 'put' },
+  { label: 'delete', value: 'delete' }
+]
+
+const memoryOptions = [
+  { label: 'Temp', value: 'temp' },
+  { label: 'Session', value: 'session' },
+  { label: 'Bot', value: 'bot' },
+  { label: 'User', value: 'user' }
+]
+
+export class CallAPI extends React.Component {
+  state = {
+    selectedMethod: methodOptions[0],
+    selectedMemory: memoryOptions[0],
+    variable: 'response',
+    body: undefined,
+    url: undefined
+  }
+
+  componentDidMount() {
+    const data = this.props.initialData
+    if (data) {
+      this.setState({
+        selectedMethod: { value: data.method, label: data.method },
+        selectedMemory: { value: data.memory, label: data.memory },
+        variable: data.variable,
+        body: data.body,
+        url: data.url
+      })
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.url && this.state.selectedMethod) {
+      const { selectedMethod, selectedMemory, body, url, variable } = this.state
+      const data = {
+        method: selectedMethod.value,
+        memory: selectedMemory.value,
+        body,
+        url,
+        variable
+      }
+
+      this.props.onDataChanged && this.props.onDataChanged(data)
+      this.props.onValidChanged && this.props.onValidChanged(true)
+    }
+  }
+
+  handleMethodChange = option => {
+    this.setState({ selectedMethod: option })
+  }
+
+  handleBodyChange = event => {
+    this.setState({ body: event.target.value })
+  }
+
+  handleURLChange = event => {
+    this.setState({ url: event.target.value })
+  }
+
+  handleMemoryChange = option => {
+    this.setState({ selectedMemory: option })
+  }
+
+  handleVariableChange = event => {
+    this.setState({ variable: event.target.value })
+  }
```

Comment:
When updating multiple inputs, I suggest making it dynamic to avoid repetition:

```js
handleInputChanged = event => this.setState({ [event.target.name]: event.target.value })

// Then for your input
// <input type="text" name="variable" value={this.state.variable}>
```